sympy/polys/constructor.py

```diff
@@ -12,13 +12,20 @@
 
 def _construct_simple(coeffs, opt):
     """Handle simple domains, e.g.: ZZ, QQ, RR and algebraic domains. """
+    from sympy.core.numbers import NaN
+    from .polyerrors import PolynomialError
+
     result, rationals, reals, algebraics = {}, False, False, False
 
     if opt.extension is True:
         is_algebraic = lambda coeff: coeff.is_number and coeff.is_algebraic
     else:
         is_algebraic = lambda coeff: False
 
+    if any(isinstance(coeff, NaN) for coeff in coeffs):
```

Comment:
I think that these tests could be done in the loop below where each coefficient is checked (rational, float or algebraic). The test could be the same as in polyutils: `if coeff in (S.NaN, S.Infinity, S.NegativeInfinity, S.ComplexInfinity)`. (Those objects are singletons so it is not necessary to use `isinstance`.) It could be the last test before returning `None` as those objects are normally not expected.